RF24.cpp

```diff
@@ -1626,12 +1643,14 @@ void RF24::openReadingPipe(uint8_t child, uint64_t address)
 
     if (child <= 5) {
         // For pipes 2-5, only write the LSB
-        if (child < 2) {
-            write_register(pgm_read_byte(&child_pipe[child]), reinterpret_cast<const uint8_t*>(&address), addr_width);
-        }
-        else {
+        if (child > 1) {
             write_register(pgm_read_byte(&child_pipe[child]), reinterpret_cast<const uint8_t*>(&address), 1);
         }
+        // avoid overwriting the TX address on pipe 0 while still in TX mode.
+        // NOTE, the cached RX address on pipe 0 is written when startListening() is called.
+        else if (static_cast<bool>(config_reg & _BV(PRIM_RX)) || child != 0) {
```

Comment:
I'm second guessing the logic here. This was meant to be a consolidated way of doing:
```cpp
if (child > 1) { /* write 1 byte from address */ }
else if (child) { /* write address to pipe 1 */ }
else if (config_reg & _BV(PRIM_RX)) { /* write address to pipe 0 */ }
else { /* skip changing pipe 0 address in TX mode */ }
```

Reply:
maybe we could try
```cpp
        if (child != 0) {
            write_register(
                pgm_read_byte(&child_pipe[child]),
                reinterpret_cast<const uint8_t*>(&address),
                child > 1 ? 1 : addr_width);
        }
        // avoid overwriting the TX address on pipe 0 while still in TX mode.
        // NOTE, the cached RX address on pipe 0 is written when startListening() is called.
        else if (static_cast<bool>(config_reg & _BV(PRIM_RX))) {
            write_register(
                pgm_read_byte(&child_pipe[child]),
                reinterpret_cast<const uint8_t*>(&address),
                addr_width);
        }
```
Although, `child > 1 ? 1 : addr_width` might be a bit costly.

Reply:
Why are you second guessing it? Is something not working?

On quick review, the logic seems to make sense.

Reply:
> Why are you second guessing it? Is something not working?

The inability to switch roles reported in #1042